runtime/compiler/compile/J9SymbolReferenceTable.cpp

```diff
@@ -1428,10 +1428,10 @@ J9::SymbolReferenceTable::findJavaLangReferenceReferentShadowSymbol(TR_ResolvedM
    TR::SymbolReference * symRef;
    TR_SymRefIterator i(type == TR::Address ? aliasBuilder.addressShadowSymRefs() :
                                             (type == TR::Int32 ? aliasBuilder.intShadowSymRefs() : aliasBuilder.nonIntPrimitiveShadowSymRefs()), self());
-   while (symRef = i.getNext())
-      if (symRef->getSymbol()->getDataType() == type &&
-          symRef->getOffset() == offset &&
-          symRef->getOwningMethod(comp()) == owningMethod)
+   while ((symRef = i.getNext()) != NULL)
+      if ((symRef->getSymbol()->getDataType() == type) &&
+          (symRef->getOffset() == offset) &&
+          (symRef->getOwningMethod(comp()) == owningMethod))
```

Comment:
Why do we need the ()s around the == components of a straight && - there are dozens of examples of && chains like this in the compiler so my preference would be to keep them uniform unless there is a real need to change them to satisfy a compiler. If we do decide to change them I'd prefer to try and make it uniformly at once to avoid mixing styles..

To be clear I'm 100% onboard with the assignment != NULL change on the while.

Reply:
I can revert that part of the change if you feel strongly about it.  

From the VM side of the code base, we have a general rule that each clause of a compound condition should be bracketed so devs can think about the logic rather than the order-of-operations, and so that future changes to the conditionals are clear when they need to introduce new sub-conditionals.

I've tried a couple of times to introduce bracketing around complex conditionals after the fact - including using tools - and had the process fail.  There's even an old, now closed PR where matching the brackets to what the clang tool says it does with the conditional results in crashes at runtime.  Easier to introduce the brackets when simple than to retrofit later.

Reply:
I certainly see there are benefits to explicit braces, but I also believe there is a benefit to keeping the code uniform. Rather than changing the code convention piecemeal I'd prefer to make this kind of change as a more considered more global change. To be clear, I am not saying you are wrong, but the conditions in this (and other compiler files) adhere to a set of conventions and I think consistency is a great aid to maintainability and readability. The benefits of your proposal should be considered as and when we get back to code style discussions.